src/rustdoc-cache.ts

```diff
@@ -61,32 +61,28 @@ export class RustdocCache {
     }
 
     private async getRustcVersion(): Promise<string> {
-        return await this.keyFromCommandCall("rustc", ["--version"]);
+        const stdout = { s: "" };
+        await exec.exec("rustc", ["--version"], this.makeExecOptions(stdout));
+        return this.removeWhitespaces(stdout.s);
     }
 
     private async getCargoSemverChecksVersion(): Promise<string> {
-        return await this.keyFromCommandCall(this.cargo, ["semver-checks", "--version"]);
+        const stdout = { s: "" };
+        await this.cargo.call(["semver-checks", "--version"], this.makeExecOptions(stdout));
+        return this.removeWhitespaces(stdout.s);
```

Comment:
Consider lifting functions like `makeExecOptions()` and `removeWhitespaces()` out of the class and just making them free functions in the module. They don't ever use `this`, and not having access to it makes them simpler to understand, less bug-prone in the future, and probably easier to optimize for the runtime as well.

Reply:
Good idea, changed!